sdk/core/azure-core/azure/core/paging/_paging_method.py

```diff
@@ -0,0 +1,241 @@
+
+    def get_continuation_token(self, pipeline_response, deserialized, continuation_token_location=None):
+        # type: (HttpResponse, ResponseType, Optional[str]) -> Any
+        """Get the continuation token from the current page. This operation returning None signals the end of paging.
+        Continuation token can be mutated here as well.
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param object deserialized: The deserialized pipeline_response
+        :param optional[str] continuation_token_location: Property name on the response object that houses the
+         continuation token. Defaults to `None`, as `None` is an acceptable value for a continuation token location.
+         It means that there's no continuation token on the response object.
+        :return: The continuation token. Can be any value, and will be passed as-is to :func:`get_next_request`.
+        :rtype: any
+        """
+
+        raise NotImplementedError("This method needs to be implemented")
+
+class ContinueWithCallback(PagingMethodABC):
+
+    def __init__(self, next_request_callback, **kwargs):  # pylint: disable=unused-argument
+        """Base paging method. Accepts the callback for the next request as an init arg.
+
+        :param callable next_request_callback: Takes the continuation token as input and
+         outputs the next request
+        """
+        self._next_request_callback = next_request_callback
+
+    def get_next_request(self, continuation_token, initial_request, client):
+        # type: (Any, HttpRequest, PipelineClient) -> HttpRequest
+        """Return the next request object for paging. Passes the continuation token into
+        the next request callback, and returns the resulting next request
+
+        :param any continuation_token: The token used to continue paging
+        :param initial_request: The initial paging request. You can use this as a foundation
+         to build up your next request
+        :type initial_request: ~azure.core.pipeline.transport.HttpRequest
+        :return: Next request for the pager to make
+        :rtype: ~azure.core.pipeline.transport.HttpRequest
+        """
+        return self._next_request_callback(continuation_token)
+
+    def get_list_elements(self, pipeline_response, deserialized, item_name="value"):
+        # type: (HttpResponse, ResponseType, str) -> Iterable[ReturnType]
+        """Extract the list elements from the current page to return to users
+
+        :param pipeline_response: The immediate response returned from the pipeline
+        :type pipeline_response: ~azure.core.pipeline.transport.HttpResponse
+        :param object deserialized: The deserialized pipeline_response
+        :return: The iterable of items extracted out of paging response
+        :rtype: iterable[object]
+        """
+        if hasattr(deserialized, item_name):
```

Comment:
Nit: you can simply try to access the name and catch AttributeError. Minor performance gain (no need to look up the attribute twice.